packages/graphin-components/src/components/legend/index.tsx

```diff
@@ -16,21 +16,24 @@ export interface LegendProps {
   onChange?: (checked: LegendOption, newOptions: LegendOption[], props: any) => any; // eslint-disable-line
 }
 
-const Legend: React.FunctionComponent<LegendProps> = props => {
-  const { onChange = () => {} } = props;
+const Legend: React.FunctionComponent<LegendProps> = (props) => {
+  const { onChange = () => {}, options: originalOptions } = props;
   // eslint-disable-next-line react/destructuring-assignment
-  const mergedOptions = props.options.map(c => {
-    const { checked } = c;
-    return {
-      ...c,
-      checked: typeof checked === 'boolean' ? checked : true,
-    };
-  });
-  const [options, setOptions] = React.useState(mergedOptions);
 
+  const mergedOptions: { (options: LegendOption[]): LegendOption[] } = (options) =>
+    options.map((c) => {
+      const { checked } = c;
+      return {
+        ...c,
+        checked: typeof checked === 'boolean' ? checked : true,
+      };
+    });
+  const [options, setOptions] = React.useState(mergedOptions(originalOptions));
+
+  useEffect(() => setOptions(originalOptions), [originalOptions]);
```

Comment:
Thank you for discovering this problem, and PR to solve it.
Is this supposed to be
`useEffect(() => setOptions(mergedOptions(props.options)), [props.options]);`

Reply:
Yes exactly. There was a eslint rule which was preventing me to use props.options and asking for de-structuring so I used new meaningful variable. 

But I missed calling mergedOptions.. my bad. Let me send an update. Thank you for pointing it out.

Reply:
@pomelo-nwu  Changes are done. Please review.

Thanks.

Reply:
nice～